### PR TITLE
Use consistent error handling.

### LINKED ISSUE #1458: Use consistent error handling.
**Issue summary**

Error handling is partially done with `Box<dyn StdError>` and partially done with `anyhow::Error`. We should always use the latter error type rather than the former.

**Other information and links**

### PR DESCRIPTION
**Pull request by tyshko5**
**Summary of changes**
Changes introduced in this pull request:
- 

**Reference issue to close (if applicable)**

Closes #1458

**Other information and links**

### CODE REVIEW

File: blockchain/state_manager/src/lib.rs [hunk 1]
```diff
@@ -98,12 +98,17 @@ where
     pub async fn new(
         cs: Arc<ChainStore<DB>>,
         chain_config: Arc<ChainConfig>,
-    ) -> Result<Self, Box<dyn std::error::Error>> {
-        let genesis = cs.genesis()?.ok_or("genesis header was none")?;
+    ) -> Result<Self, anyhow::Error> {
+        let genesis = cs
+            .genesis()
+            .map_err(|e| anyhow::anyhow!("{}", e))?
```
Comment (lemmih) [inline blockchain/state_manager/src/lib.rs:104]: This `map_err` isn't necessary.

**Review by lemmih — CHANGES_REQUESTED** [review]
Never convert an error to a string unless you have to. Remove all the `map_err`s that you can.

**Review by lemmih — CHANGES_REQUESTED** [review]

File: utils/net_utils/src/download.rs [hunk 2]
```diff
@@ -65,7 +65,7 @@ impl TryFrom<Url> for FetchProgress<AsyncBody, Stdout> {
                     .and_then(|ct_len| ct_len.parse().ok())
                     .unwrap_or(0)
             } else {
-                return Err(Box::new(DownloadError::HeaderError));
+                return Err(anyhow::anyhow!("{}", DownloadError::HeaderError));
```
[the remark this answers is not in the document]
Comment (tyshko5) [reply utils/net_utils/src/download.rs:68]: It isn't going to be a string

[the remark this answers is not in the document]
Comment (lemmih) [reply utils/net_utils/src/download.rs]: Could you explain that to me?

[the remark this answers is not in the document]
Comment (tyshko5) [reply utils/net_utils/src/download.rs]: But I've changed it to ::new if that's more acceptable

[the remark this answers is not in the document]
Comment (lemmih) [reply utils/net_utils/src/download.rs]: `anyhow::anyhow!(DownloadError::HeaderError)` would also have been acceptable.

**Review by lemmih — APPROVED** [review]

**Comment by lemmih** [comment]
> What about using `.context` instead of `.ok_or_else(|| ..`?

Ah, yes. Much better. I thought `.context` was only defined for `Result` and not `Option`.

File: blockchain/state_manager/src/lib.rs [hunk 3]
```diff
@@ -98,8 +98,10 @@ where
     pub async fn new(
         cs: Arc<ChainStore<DB>>,
         chain_config: Arc<ChainConfig>,
-    ) -> Result<Self, Box<dyn std::error::Error>> {
-        let genesis = cs.genesis()?.ok_or("genesis header was none")?;
+    ) -> Result<Self, anyhow::Error> {
+        let genesis = cs
+            .genesis()?
+            .ok_or_else(|| anyhow::anyhow!("genesis header was none"))?;
```
Comment (LesnyRumcajs) [inline blockchain/state_manager/src/lib.rs:104]: perhaps `geneis header missing` would be more helpful for those not versed in Rust scriptures

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**